lib/core/src/recover/recoverer.rs

```diff
@@ -161,11 +161,19 @@ impl Recoverer {
                         log::warn!("Could not apply recovered data for Send swap {swap_id}: recovery data not found");
                         continue;
                     };
-                    let is_expired = liquid_height
-                        >= send_swap.get_boltz_create_response()?.timeout_block_height as u32;
+                    let timeout_block_height =
+                        send_swap.get_boltz_create_response()?.timeout_block_height as u32;
+                    let is_expired = liquid_tip.height >= timeout_block_height;
                     if let Some(new_state) = recovered_data.derive_partial_state(is_expired) {
                         send_swap.state = new_state;
                     }
+                    send_swap.expiry_at = self
```

Comment:
Can we estimate the expiration using the current tip timestamp and the timeout block also for expired swap?
It will save us the interface change to the chain service and an extra call for the block header.

Reply:
I've decided to go for the interface change since it looked neater, but technically it can be optimized by simply moving these methods to the recoverer and reusing the tip. See 71896e696d43

Reply:
What I meant is that we can use the tip timestamp (as introduced previously by Ross) to calculate also expired swap expiry.
`(tip timestamp) - (tip_height - timeout_block_height) * BLOCK_FREQUENCY`

This way we don't need the new methods.

Reply:
Well, my understanding from speaking to Ross was that we should use the actual timestamp of the block (since it's already passed) rather than an estimation, and honestly I agree with that (especially on the Bitcoin side, where mining times are less predictable).
@dangeross do you agree?

Reply:
TBH the expiration is only required for the pending case. I mentioned that it would be nice to have it for all payments just for consistency.
In the pending case (and when we create the swap) we anyway put an estimation so we won't be able to set the exact block timestamp in the common cases.

Reply:
I see your point. I agree the extra overhead may not be worth it, so I guess an estimation is fine

Reply:
I was thinking more about how to set the expiry for chain swaps that we continue to monitor with the recoverer regardless of its state up to a month after the expiry. Should we estimate then or just take the actual block time

Reply:
I would also note that if the swap changes (because of the expiry_at changes) an event is emitted